core/utils/util.c

```diff
@@ -33,12 +33,29 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include "util.h"
 
+#if !defined(PLATFORM_FLEXPRET)
+#include <stdio.h>
+#else
+/**
+ * For FlexPRET specifically, a small footprint version of printf is used
+ */
+#include <printf/printf.h>
+/**
+ * Also, fflush and stdout are not provided so we just implement stubs
+ *
+ */
+#define stdout (void*)(1)
+int fflush(void* stream) {
+  (void)stream;
+  return 0;
+}
+#endif // PLATFORM_FLEXPRET
+
```

Comment:
I am not too excited about putting this here. Ideally, everything that is platform-specific should be under low-level-platform.  What happens if you include stdio for FlexPRET, do we get linker errors with the printf lib? 

If we really cannot have stdio.h for FlexPRET, I propose that we move this to platform.h, i.e. lf_platform_flexpret.h includes printf/printf.h, and the other platform files include stdio. ALso this redefinition of stdout and fflush I would put in lf_platform_flexpret.h/c.

Reply:
Right, very good point. I'll move it to low level platform. I think I got some errors when #included stdio.h, which is the reason I hid it behind the #if. But I'll look into if I can fix it so #include stdio does not give compile errors.

Reply:
The underlying issue I had was that I got this error message:

```
/opt/riscv/riscv32-unknown-elf/include/stdio.h:207:16: error: 'format' attribute argument 2 value '2' refers to parameter type '__gnuc_va_list' {aka 'void *'}
  207 |                _ATTRIBUTE ((__format__ (__printf__, 2, 0)));
```

Which is the reason I tried not to include `stdio.h` in the source files. However, the solution was to just include it before these two lines:

```
#define fprintf(stream, fmt, ...) printf(fmt, ##__VA_ARGS__)
#define vfprintf(fp, fmt, args) vprintf(fmt, args)
```

That makes sense, because if `vfprintf` is #defined to `vprintf` (same just without an "f"), and `stdio.h` is included after, its declaration in the `stdio.h` file will go from:

```
int	vfprintf (FILE *__restrict, const char *__restrict, __VALIST)
               _ATTRIBUTE ((__format__ (__printf__, 2, 0)));
```

to 

```
int	vprintf (FILE *__restrict, const char *__restrict, __VALIST)
               _ATTRIBUTE ((__format__ (__printf__, 2, 0)));
```

which I guess should yield an error, because `vprintf`'s actual declaration looks like this:

```
int	vprintf (const char *, __VALIST)
               _ATTRIBUTE ((__format__ (__printf__, 1, 0)));
```

But since it now is included before these #defines, it works just fine. So I've removed the platform-specific changes all together :)